root/static/scripts/edit/URLCleanup.js

```diff
@@ -1920,12 +1920,17 @@ const CLEANUPS = {
     match: [new RegExp('^(https?://)?([^/]+\\.)?songkick\\.com', 'i')],
     type: LINK_TYPES.songkick,
     clean: function (url) {
-      return url.replace(/^http:\/\//, 'https://');
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?songkick\.com\//, 'https://www.songkick.com/');
+      url = url.replace(/^(https:\/\/www\.songkick\.com\/[a-z]+\/[0-9]+)(?:-[\w-]*)?(\/id\/[0-9]+)?(?:[-\/?#].*)?$/, '$1$2');
+      return url;
     },
     validate: function (url, id) {
-      const m = /songkick\.com\/([a-z]+)\//.exec(url);
+      const m = /^https:\/\/www\.songkick\.com\/([a-z]+)\/[0-9]+(?:\/(id)\/[0-9]+)?$/.exec(url);
       if (m) {
         const prefix = m[1];
+        if ((m[2] === 'id') !== (prefix === 'festivals')) {
```

Comment:
I'm not sure I'm following this one. Is this supposed to be "if m[2] is id then prefix should be festivals"? If so, isn't there a more legible way of writing that? :/

Reply:
It is more than that: _"if m[2] is id then prefix should be festivals"_ and "if prefix is festivals then m[2] should be id", that is "m[2] is id **if and only if** prefix is festivals", which is negated here with central `!`.

Reply:
If this is the easiest way to say this, then let it be :)

Reply:
It's basically an XOR, which JavaScript doesn't have an operator for, so it's at least the most compact way to write it. The alternative is `if ((m[2] === 'id' && prefix !== 'festivals') || (m[2] !== 'id' && prefix === 'festivals')) return false` but that's not much clearer to me.